Jint/ParseOptions.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System.Text.RegularExpressions;
+
+namespace Jint
+{
+    public interface IParseOptions
+    {
+        /// <summary>
+        /// Gets or sets whether to create compiled <see cref="Regex"/> instances when adapting regular expressions.
+        /// Defaults to <see langword="null"/>, which means that in the case of non-prepared scripts and modules
+        /// regular expressions will be interpreted, otherwise they will be compiled.
+        /// </summary>
+        bool? CompileRegex { get; init; }
+
+        /// <summary>
+        /// Gets or sets the default timeout for created <see cref="Regex"/> instances.
+        /// Defaults to <see langword="null"/>, which means that in the case of non-prepared scripts and modules
+        /// the <see cref="Options.ConstraintOptions.RegexTimeout"/> setting should apply,
+        /// otherwise the default of the <see cref="ParserOptions.RegexTimeout"/> setting (10 seconds).
+        /// </summary>
+        /// <remarks>
+        /// Please note that <see cref="Options.ConstraintOptions.RegexTimeout"/> setting will be ignored
+        /// if this option is set to a value other than <see langword="null"/>.
+        /// </remarks>
+        TimeSpan? RegexTimeout { get; init; }
+
+        /// <summary>
+        /// Gets or sets whether to parse the source code in tolerant mode.
+        /// Defaults to <see langword="true"/>.
+        /// </summary>
+        /// <remarks>
+        /// Please note that there is a plan to change the default to <see langword="false"/> in the next major version.
+        /// </remarks>
+        bool Tolerant { get; init; }
+    }
+
+    public record class ScriptParseOptions : IParseOptions
+    {
+        private static readonly ParserOptions _defaultParserOptions = Engine.BaseParserOptions with
+        {
+            AllowReturnOutsideFunction = true,
+            RegExpParseMode = RegExpParseMode.AdaptToInterpreted,
+        };
+
+        public static readonly ScriptParseOptions Default = new();
+
+        /// <summary>
+        /// Gets or sets whether to allow return statements at the top level.
+        /// Defaults to <see langword="true"/>.
+        /// </summary>
+        /// <remarks>
+        /// Please note that there is a plan to change the default to <see langword="false"/> in the next major version.
```

Comment:
next major of Acornima? so should this just be `true`?

Reply:
I meant the next major version of Jint.

Or we could just make this BC in v3.1 as the related APIs will be broken anyway. What do you think?

Reply:
I'm not just quite sure if it's best approach on interpreter side to be strict-by-default. For parsing it makes sense, but for anyone just "having a good time with JS" do we need to restrict parsing and allowed constructs?  But again seems that NodeJS throws error on console from using return so maybe it makes sense.

Reply:
I wrote about this in the other PR: "[...] I can only speak for myself, but I'd be surprised if I fed some script to Jint which is invalid per spec and get some results instead of SyntaxError. I mean I think relaxing the rules should be a deliberate choice, not the default. [...]"

> But again seems that NodeJS throws error on console from using return so maybe it makes sense.

Disabling tolerant mode doesn't affect top level return because that's controlled by another switch than tolerant mode. The option would be on for scripts by default: https://github.com/sebastienros/jint/pull/1831/files#diff-54aceb67398ce1901835b606789cff7f0ac2abb5136a75480469b5c732d19b11R40

Reply:
Let's not break this behavior in a minor version, so I'd say let's leave it as is now and reconsider it when v4 is on the table.